### PR TITLE
adding profile page

### PR DESCRIPTION
**Pull request by marinaRomanchuk**

### CODE REVIEW

File: src/components/User.js [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+                post.showComments = false;
+            }
+            setPostList({postList: posts});
+            setLoading(false);
+        }
+        handleSome();
+    }, []);
+
+
+    const clickLike = (postId, index) => {
```
Comment (revimi) [inline src/components/User.js]: For this and other functions: let's not copy-paste them but put into a common place to follow DRY style

Comment (marinaRomanchuk) [reply src/components/User.js]: Done

File: src/components/Functions.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        .then(result => result.data);
+}
+
+export async function getSelfInfo(authStr) {
+    return axios.get("http://" + API_DOMAIN + "/api/users/me/", { headers: { Authorization: authStr }})
+        .then(result => result.data);
+}
+
+export async function getUserInfo(authStr, userId) {
+    const userStr = ''.concat(userId);
```
Comment (revimi) [inline src/components/Functions.js]: What does it do?

File: src/components/Functions.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
+import axios from "axios";
```
Comment (revimi) [inline src/components/Functions.js]: We may probably have structure like this:
`src/lib/api/` to store functions that directly call the API
`src/lib/components/` to store reused components
`src/pages/` to store the pages (for example, `profile.js`, `feed.js`)

File: src/components/Functions.js (hunk 2 above)
Comment (marinaRomanchuk) [reply src/components/Functions.js]: I tried to make string from int, removed

File: src/components/Functions.js (hunk 3 above)
Comment (marinaRomanchuk) [reply src/components/Functions.js]: Done

File: src/lib/api/RequestFunctions.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
+import axios from "axios";
```
Comment (revimi) [inline src/lib/api/RequestFunctions.js:1]: We can also split this file into 
`posts.js`
`users.js`
and so on

Comment (revimi) [inline src/lib/api/RequestFunctions.js]: We can also split this long file into shorter and more readable:
`posts.js`
`users.js`
and so on

File: src/lib/components/Components.js [hunk 5]
```diff
@@ -0,0 +1,207 @@
+import {
```
Comment (revimi) [inline src/lib/components/Components.js]: I would also split this file into 
`likes`
`posts`
`users`
etc

Comment (marinaRomanchuk) [reply src/lib/components/Components.js]: Done

File: src/lib/api/RequestFunctions.js (hunk 4 above)
Comment (marinaRomanchuk) [reply src/lib/api/RequestFunctions.js]: Done

File: src/lib/api/Posts.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    return axios.get("http://" + API_DOMAIN + "/api/posts/", { headers: { Authorization: authStr },
+        params: { feed: true,}})
+        .then(result => result.data);
+}
+
+export function postPost(authStr, credentials) {
+    return axios.post("http://" + API_DOMAIN + "/api/posts/create/", credentials, {
+        headers: {Authorization: authStr}
+    }).catch((error) => {
+        console.log(error.response);
```
Comment (revimi) [inline src/lib/api/Posts.js:21]: We probably need to show correct error message to the user. For now, it is not necessary, but it should be updated in the future. So let's add a `todo` here and in other places were we just log the error

**Review by revimi — APPROVED** [review]
:+1:

File: src/lib/api/Posts.js (hunk 6 above)
Comment (marinaRomanchuk) [reply src/lib/api/Posts.js:21]: Ok

**Pull request merged**